Source/items.cpp

```diff
@@ -1741,76 +1741,98 @@ void DrawUniqueInfoWindow(const Surface &out)
-			} else if (!invflag) {
-				AddPanelString(_("Open inventory to use"));
-			} else {
-				AddPanelString(_("Activate to read"));
-			}
-		}
+	if (item._iMiscId == IMISC_MAPOFDOOM) {
+		AddPanelString(_("Right-click to view"));
+
+	} else if (isOil) {
+		PrintItemOil(item._iMiscId);
+		AddPanelString(_("Right-click to use"));
+	} else if (isCastOnTarget) {
+		AddPanelString(_("Right-click to read, then"));
+		AddPanelString(_("left-click to target"));
+	} else if (IsAnyOf(item._iMiscId, IMISC_BOOK, IMISC_NOTE, IMISC_SCROLL)) {
+		AddPanelString(_("Right-click to read"));
 	}
-	if ((item._iMiscId >= IMISC_USEFIRST && item._iMiscId <= IMISC_USELAST)
-	    || (item._iMiscId > IMISC_OILFIRST && item._iMiscId < IMISC_OILLAST)
-	    || (item._iMiscId > IMISC_RUNEFIRST && item._iMiscId < IMISC_RUNELAST)) {
+}
+
+void printItemMiscVirtualGamepad(const Item &item, const bool isOil)
+{
+	if (item._iMiscId == IMISC_MAPOFDOOM) {
+		AddPanelString(_("Activate to view"));
+	} else if (isOil) {
 		PrintItemOil(item._iMiscId);
-		if (ControlMode == ControlTypes::KeyboardAndMouse) {
-			AddPanelString(_("Right-click to use"));
+		if (!invflag) {
+			AddPanelString(_("Open inventory to use"));
+
 		} else {
-			if (!invflag) {
-				AddPanelString(_("Open inventory to use"));
-			} else {
-				AddPanelString(_("Activate to use"));
-			}
+			AddPanelString(_("Activate to use"));
 		}
+	} else if (item._iMiscId == IMISC_SCROLL) {
+		AddPanelString(_("Select from spell book, then"));
+		AddPanelString(_("cast to read"));
+	} else {
+		AddPanelString(_("Activate to read"));
 	}
-	if (IsAnyOf(item._iMiscId, IMISC_BOOK, IMISC_NOTE)) {
-		if (ControlMode == ControlTypes::KeyboardAndMouse) {
-			AddPanelString(_("Right-click to read"));
-		} else {
-			AddPanelString(_("Activate to read"));
-		}
+}
+
+void printItemMiscGamepad(const Item &item, bool isOil, bool isCastOnTarget)
+{
+	std::string activateButton = "Activate";
+	std::string castButton = "Cast";
+
+	if (GamepadType == GamepadLayout::Xbox) {
+		activateButton = "Y";
+		castButton = "X";
+	} else if (GamepadType == GamepadLayout::PlayStation) {
+		activateButton = "Triangle";
+		castButton = "Square";
+	} else if (GamepadType == GamepadLayout::Nintendo) {
+		activateButton = "Y";
+		castButton = "X";
 	}
+
 	if (item._iMiscId == IMISC_MAPOFDOOM) {
-		if (ControlMode == ControlTypes::KeyboardAndMouse) {
-			AddPanelString(_("Right-click to view"));
+		AddPanelString(_(fmt::format("{} to view", activateButton)));
+	} else if (isOil) {
+		PrintItemOil(item._iMiscId);
+		if (!invflag) {
+			AddPanelString(_("Open inventory to use"));
 		} else {
-			AddPanelString(_("Activate to view"));
+			AddPanelString(_(fmt::format("{} to use", activateButton)));
 		}
+	} else if (isCastOnTarget) {
+		AddPanelString(_(fmt::format("Select from spell book, then {} to read", castButton)));
+	} else if (IsAnyOf(item._iMiscId, IMISC_BOOK, IMISC_NOTE, IMISC_SCROLL)) {
+		AddPanelString(_(fmt::format("{} to read", activateButton)));

```

Comment:
You translate the string litteral, not the resulting string. Else translators would have to predict the future when making translations :)

```suggestion
		AddPanelString(fmt::format(_("{} to read", activateButton)));

```